src/gdal/ogr_file_format.cpp

```diff
@@ -1408,5 +1416,354 @@ LatLon OgrFileImport::calcAverageLatLon(OGRDataSourceH data_source)
+		OSRSetWellKnownGeogCS(map_srs.get(), "WGS84");
+		auto spec = QByteArray(georef.getProjectedCRSSpec().toLatin1() + " +wktext");
+		if (OSRImportFromProj4(map_srs.get(), spec) != OGRERR_NONE)
+		{
+			local_only = true;
+			addWarning(tr("Failed to properly create the export georeferencing info.  Local georeferencing only."));
+		}
+	}
+
+	/**
+	 * Only certain drivers work without georeferencing info
+	 * Based on http://www.gdal.org/ogr_formats.html as of March 5, 2018
+	 * */
+	static const std::vector<QByteArray> local_drivers = {
+	    "ARCGEN", "BNA", "DWG", "DXF",
+	    "DXF", "CSV", "Geomedia", "INGRES",
+	    "OpenJUMP .jml", "DGN", "DGNv8", "ODS",
+	    "REC", "SEGY", "XLS", "XLSX"
+	};
+	if (local_only)
+	{
+		QByteArray driver_name = GDALGetDriverLongName(po_driver);
+		if (std::find(local_drivers.begin(), local_drivers.end(), driver_name) == local_drivers.end())
+		throw FileFormatException(tr("The %1 driver requires valid georefencing info.")
+		                          .arg(QString::fromLatin1(driver_name)));
+	}
+
+	// Create output dataset
+	po_ds = ogr::unique_datasource(OGR_Dr_CreateDataSource(po_driver, output_file.toLatin1(), nullptr));
+	if (!po_ds)
+		throw FileFormatException(tr("Failed to create dataset.  Error: %1").arg(QString::fromLatin1(CPLGetLastErrorMsg())));
+
+	/**
+	 * Create collections of point, line, and areas that are selected
+	 * These have to be separate as some drivers required them to be added in a specific order
+	 */
+	auto num_parts = map->getNumParts();
+	for (auto i = 0; i < num_parts; i++)
+	{
+		auto part = map->getPart(i);
+		auto num_objects = part->getNumObjects();
+		for (auto j = 0; j < num_objects; j++)
+		{
+			auto object = part->getObject(j);
+			switch (object->getSymbol()->getType())
+			{
+			case Symbol::Point :
+				point_objects.push_back(object);
+				break;
+			case Symbol::Text :
+				text_objects.push_back(object);
+				break;
+			case Symbol::Line :
+				line_objects.push_back(object);
+				break;
+			case Symbol::Area :
+			case Symbol::Combined :
+				area_objects.push_back(object);
+				break;
+			default:
+				break;
+			}
+		}
+	}
+
+	// For GPX driver, coords have to be in EPSG:4326/WGS 1984
+	auto geo_srs = ogr::unique_srs { OSRNewSpatialReference(nullptr) };
+	OSRSetWellKnownGeogCS(geo_srs.get(), "WGS84");
+	transformation = ogr::unique_transformation { OCTNewCoordinateTransformation(map_srs.get(), geo_srs.get()) };
+
+	// Name field definition
+	o_name_field = ogr::unique_fielddefn(OGR_Fld_Create("Name", OFTString));
+	OGR_Fld_SetWidth(o_name_field.get(), 32);
+
+	if (point_objects.size() > 0)
+		createPointLayer();
+
+	if (text_objects.size() > 0)
+		createTextLayer();
+
+	if (line_objects.size() > 0)
+		createLineLayer();
+
+	if (area_objects.size() > 0)
+		createAreaLayer();
+}
+
+void OgrFileExport::createPointLayer()
+{
+	auto po_point_layer = GDALDatasetCreateLayer(po_ds.get(), "point_symbols", map_srs.get(), wkbPoint, nullptr);
```

Comment:
In GDAL/OGR 1.x, we don't have `GDALDatasetCreateLayer()` but must use `OGR_DS_CreateLayer()`. There are multiple occurences of `GDALDatasetCreateLayer()`.

OGR 1.11 API documentation is here: http://gdal.org/1.11/ogr/index.html